### PR TITLE
Fix arena tags not being applied on session load

### PR DESCRIPTION
**Pull request by happinyz**
This PR addresses a reported issue where hazards were being removed when loading a save session: https://github.com/pagefaultgames/pokerogue/issues/684
I've verified that this fix works for other Arena Tags, as well as loading a save from the 'Load Game' option.

Stealth rock:

https://github.com/pagefaultgames/pokerogue/assets/59787978/41bacb1c-acd4-499c-9941-2f82ec5583ae

Spikes/Toxic Spikes:

https://github.com/pagefaultgames/pokerogue/assets/59787978/6b2bcd2d-4489-4e4f-a557-6b141e1cb938

Wish: 

https://github.com/pagefaultgames/pokerogue/assets/59787978/a4c03425-8019-4ccb-92cd-97dfae252a9d

### CODE REVIEW

**Comment by Xavion3** [comment]
I doubt this is going to work, so I'll just ask about some of the cases where I expect it not to work.

Has this been tested with multiple layers of spikes to ensure it maintains the correct amount of layers? Toxic spikes has layers too that could be tested.
Has this been tested with wish to ensure it still heals and heals the correct amount?

You can look at what I did with battlerTag save/loading for a reference if you want, but it wound up needing significantly more than this.

**Comment by happinyz** [comment]
> I doubt this is going to work, so I'll just ask about some of the cases where I expect it not to work.
> 
> Has this been tested with multiple layers of spikes to ensure it maintains the correct amount of layers? Toxic spikes has layers too that could be tested. Has this been tested with wish to ensure it still heals and heals the correct amount?
> 
> You can look at what I did with battlerTag save/loading for a reference if you want, but it wound up needing significantly more than this.

You were right, those cases failed in my initial implementation. I've updated the PR to account for these cases, and posted demo videos. Thanks for mentioning your work with the battler tags, that really helped!

**Comment by happinyz** [comment]
@Xavion3 Could you give this another look? Thanks!

File: src/data/arena-tag.ts [hunk 1]
```diff
@@ -496,7 +556,7 @@ class TailwindTag extends ArenaTag {
   }
 }
 
-export function getArenaTag(tagType: ArenaTagType, turnCount: integer, sourceMove: Moves, sourceId: integer, targetIndex?: BattlerIndex, side: ArenaTagSide = ArenaTagSide.BOTH): ArenaTag {
```
Comment (Xavion3) [inline src/data/arena-tag.ts:499]: Why are you changing the order of arguments?

File: src/data/arena-tag.ts [hunk 2]
```diff
@@ -531,3 +591,14 @@ export function getArenaTag(tagType: ArenaTagType, turnCount: integer, sourceMov
...
+/**
+  * When given an arena tag or json representing one, creates the actual ArenaTag object with the same data loaded.
+  * @param {ArenaTag | any} source An arena tag, or json representing one.
+  * @return {ArenaTag} The valid arena tag with correct data loaded.
+  */
+export function loadArenaTag(source: ArenaTag | any): ArenaTag {
+  const tag = getArenaTag(source.tagType, source.turnCount, source.sourceMove, source.sourceId, source.side, source?.targetIndex);
+  tag.loadTag(source);
+  return tag;
+}
```
Comment (Xavion3) [inline src/data/arena-tag.ts:604]: Keep ending newline consistent preferably, just increases compatibility with other PRs.

**Review by Xavion3 — CHANGES_REQUESTED** [review]
Is this tested against tags with custom private data like wish?

File: src/data/arena-tag.ts (hunk 2 above)
Comment (happinyz) [reply src/data/arena-tag.ts]: Fixed

File: src/data/arena-tag.ts (hunk 1 above)
Comment (happinyz) [reply src/data/arena-tag.ts:499]: optional params should be at the end of the parameter list

**Comment by happinyz** [comment]
> Is this tested against tags with custom private data like wish?

Yes, the main description of the PR has a demo video for Wish. Are there other specific tags you'd like me to test?

**Comment by Tempo-anon** [comment]
Closing due to inactivity and merge conflicts. If this is still required, feel free to open a new PR

**Pull request closed without merging**